spec/models/grade_entry_item_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'spec_helper'
+
+describe GradeEntryItem do
+
+  context 'checks relationships' do
+    it { is_expected.to belong_to(:grade_entry_form) }
+    it { is_expected.to have_many(:grades).dependent(:delete_all) }
+    it { is_expected.to have_many(:grade_entry_students).through(:grades) }
+
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_uniqueness_of(:name).scoped_to(
+      :grade_entry_form_id).with_message(
+      I18n.t('grade_entry_forms.invalid_name')) }
+
+    it { is_expected.to validate_presence_of(:out_of) }
+    it { is_expected.to validate_numericality_of(:out_of).with_message(
+      I18n.t('grade_entry_forms.invalid_column_out_of')) }
+
+    it { is_expected.to allow_value(0).for(:out_of) }
+    it { is_expected.to allow_value(1).for(:out_of) }
+    it { is_expected.to allow_value(2).for(:out_of) }
+    it { is_expected.to allow_value(100).for(:out_of) }
+    it { is_expected.not_to allow_value(-1).for(:out_of) }
+    it { is_expected.not_to allow_value(-100).for(:out_of) }
+
+
+    it { is_expected.to validate_presence_of(:position) }
+    it { is_expected.to validate_numericality_of(:position) }
+
+    it { is_expected.to allow_value(0).for(:position) }
+    it { is_expected.to allow_value(1).for(:position) }
+    it { is_expected.to allow_value(2).for(:position) }
+    it { is_expected.to allow_value(100).for(:position) }
+    it { is_expected.not_to allow_value(-1).for(:position) }
+    it { is_expected.not_to allow_value(-100).for(:position) }
+  end
+
+  context 'A good Grade Entry Item model' do
+    before { GradeEntryItem.create(:position => 1) }
+
+    it { is_expected.to validate_uniqueness_of(:name).scoped_to(
+      :grade_entry_form_id).with_message(
+      I18n.t('grade_entry_forms.invalid_name')) }
+  end
+
+  # Make sure different grade entry forms can have grade entry items
+  # with the same name
+  it 'allow same column name for different grade entry forms' do
+    grade_entry_form_1 = GradeEntryForm.create!(:short_identifier => 'a', :is_hidden => false)
```

Comment:
1. In general, we are switching to use the new object syntax `short_identifier: 'a'` instead, so please change this. Also, I think we can get rid of the `is_hidden` option altogether.